src/lotus/message/ipc.rs

```diff
@@ -29,3 +28,17 @@ pub struct IPCReadSubnetActorStateResponse {
     pub check_period: ChainEpoch,
     pub validator_set: ValidatorSet,
 }
+
+#[derive(Deserialize, Serialize, Debug)]
```

Comment:
You had to duplicate the structs here because there `serialize_tuple` didn't work, right? Also, double-check always the order of the fields when using `serialize_tuple`, as the fields are not the indexed by name but deterministically by their order (their underlying representation is a list)  

Do you mind adding a note here of why you need to re-write these structs?

Reply:
Yeah, comments added. But the idea is in the actor, it is `deserialize_tuple`, but when returned from json rpc endpoints, it's actually `json` struct. It's not working because the agent is interpreting the tuple as json.